cpp/ql/lib/experimental/semmle/code/cpp/semantic/analysis/RangeAnalysisImpl.qll

```diff
@@ -0,0 +1,105 @@
+private import RangeAnalysisStage
+private import RangeAnalysisSpecific
+private import experimental.semmle.code.cpp.semantic.analysis.FloatDelta
+private import RangeUtils
+private import experimental.semmle.code.cpp.semantic.SemanticBound as SemanticBound
+private import semmle.code.cpp.ir.IR as IR
+private import semmle.code.cpp.Location // TODO: SemLocation?
```

Comment:
This is also the only language-specific parts of this file, right? I think we should move it out. As far as I can see, the only place where `IR` is used is in
```ql
this.(SemanticBound::SemSsaBound).getExpr(0) instanceof IR::PhiInstruction
```
which I hope we can refactor 🤞.